### ISSUE TITLE
Flatpak package do not behave with files stored on USB devices correctly !

### ISSUE THREAD

**Issue by Nokia808** [flathub/org.nomacs.ImageLounge#5]
Hi.
I'm using fatpak version of nomacs on my Fedora 32. I removed all other image viewer (apart from GIMP because edit extensions still not available on FlatHub). I set nomacs as the default image viewer for all image formates (jpeg, png, .....).

I noticed that I can not navigate to "next" nor "previous" images on that USB device, because nomacs show me only selected image !

Can you, kindly, fix this ? I think that this is a permission problem ...

**Comment by diemmarkus** [comment]
I head a [read up](https://github.com/flatpak/flatpak/issues/2713) and think now that users should fix this on their own:
````
flatpak override --user --filesystem=/path/to/dir org.nomacs.ImageLounge
````

**Comment by Nokia808** [comment]
@diemmarkus 
Dear what "path/to/dir" refer to ? Does it refer to path of USB device ?

This is too much on end users ! Just I like to understand one thing: why Flatpak (not you) behave like blaming itself by itself ?! It behave so that it saying: "do not trust me"!! I would like to understand: why we should consider .rpm/.deb packages packagers (which almost a 3rd party human not a member in developer team of application) as Engels & flatpak packagers as crackers ?! This, currently, what flatpak design try to say all the time !!

Can you fix this globally ? What it will make flatpak dangerous more than .rpm if you fix this globally ??
If I should not trust you (developer), so why I should trust Fedora packager of nomacs which is not a developer team member ?? Repeatedly I'm asking my self this question ...

**Comment by diemmarkus** [comment]
yep `/path/to/dir` could be `/media/USERNAME`. I have now added the media folder by default (since this shouldn't do any harm). 

> It behave so that it saying: "do not trust me"!!

you know, I am new to flatpak too - so I am sure that the nomacs package has some issues because I am far from being a flatpak guru. but sandboxing is not only a matter of trusting a developer: with all my best intentions there can be security issues that malware could exploit.

**Comment by Nokia808** [comment]
@diemmarkus 
Hi. I updated to version 3.16 from FlatHub but this issue did not fixed ! It is still existing !
Kindly re-open it.

**Comment by Nokia808** [comment]
@diemmarkus 
Dear developer can you kindly to test if this issue existing with .rpm / .deb packages or not ?

Also - if it is not existing with .rpm / .deb packages - does it existing with flatpak installed on other distros that not using SELinux ? There are certain cases when SELinux cause aberrations without giving error messages ...

**Comment by diemmarkus** [comment]
domy also mentioned this issue [here](https://nomacs.org/questions-and-answers/#comment-2861)

**Comment by Nokia808** [comment]
Hi. This issue is not limited to flatpak package. I test .rpm package from official Fedora repositories & behave same way !

**Comment by Nokia808** [comment]
@diemmarkus 
Hi. I added previous comment which then I deleted it just few seconds because it contained wrong data due to a confusion from my side. And to be clear this issue is LIMITED ONLY TO FLATPAK version never to .rpm / .deb package so it is a flatpak specific.

**Comment by Hasshu** [comment]
@Nokia808 It's called the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege). Ultimately, it's up to the developers to make their Flatpak usable.

@novomesk Speaking of which, how about adding `--filesystem=/mnt` to the manifest? That should get most bases covered.

**Comment by Hasshu** [comment]
> Speaking of which, how about adding `--filesystem=/mnt` to the manifest? That should get most bases covered.

Also paging @bilelmoussaoui, @diemmarkus, @hfiguiere...
